webapp/lib/view.dart

```diff
@@ -1371,6 +1306,81 @@ class ProjectConfigurationView extends BaseView{
   }
 }
 
+abstract class BaseElement {
+  Element get renderElement;
+}
+
+class DropdownElement extends BaseElement {
+  Element _dropdown;
+
+  DropdownElement(List<String> dropdownItems, Function onItemSelected) {
+    _dropdown = new Element.ul()
+      ..classes.add('add-tag-dropdown');
+    var items = dropdownItems.isEmpty ? ['--None--'] : dropdownItems;
+    for (var item in items) {
+      _dropdown.append(
+        new Element.li()
+          ..classes.add('add-tag-dropdown__item')
+          ..text = item
+          ..dataset['item'] = item
+          ..onClick.listen(onItemSelected)
+      );
+    }
+
+    var documentOnClickSubscription;
+    documentOnClickSubscription = document.onClick.listen((event) {
+      if (event.target == document.activeElement) return;
+      event.stopPropagation();
+      _dropdown.remove();
+     documentOnClickSubscription.cancel();
+  });
+}
+
+  Element get renderElement => _dropdown;
+}
+
+class EditableTagElement extends BaseElement {
+  TagView _editableTag;
+
+  EditableTagElement(Function onEditTag) {
+    _editableTag = new TagView('', model.TagType.Normal, onEditTag, true);
+  }
+
+  void focus() {
+    _editableTag.focus();
+  }
+
+  Element get renderElement => _editableTag.renderElement;
+}
+
+class MenuElement extends BaseElement {
+  ButtonElement _menu;
+
+  MenuElement(Element container, List<String> dropdownItems, Function onItemSelected) {
+    _menu = new ButtonElement()
+      ..classes.add('active-package-menu')
+      ..text = ''
+      ..onClick.listen((_) {
+        container.append(new DropdownElement(dropdownItems, onItemSelected).renderElement);
+    });
+  }
+
+  Element get renderElement => _menu;
+}
+
+class AddButtonElement extends BaseElement {
```

Comment:
consider renaming to `AddTagButtonElement` for more specificity (and reflects the CSS class as well)

Reply:
Renamed.